specs/dwritecore/DWriteCore.md

```diff
@@ -78,33 +78,275 @@ with interoperability issues.
+font in a WWS family must have a unique location defined by its coordinates along those three axes. To
+select a font, one specifies a WWS family name and weight, stretch, and style parameters.
+
+By contrast, the typographic font family model has an N-dimensional design space. A font designer can 
+define any number of design axes, each identified by a four-character _axis tag_. A given font’s location 
+in the N-dimensional design space is defined by an array of _axis values_, where each axis value comprises 
+an axis tag and a floating-point value. To select a font, one specifies a typographic family name and an 
+array of axis values (`DWRITE_FONT_AXIS_VALUE` structures).
+
+Although the number of font axes is open-ended, there are a few registered axes with standard meanings,
+and the weight, stretch, and style values can be mapped to registered axis values. `DWRITE_FONT_WEIGHT`
+can be mapped to a "wght" (`DWRITE_FONT_AXIS_TAG_WEIGHT`) axis value. `DWRITE_FONT_STRETCH` can be mapped
+to a "wdth" (`DWRITE_FONT_AXIS_TAG_WIDTH`) axis value. `DWRITE_FONT_STYLE` can be mapped to a combination
+of "ital" and "slnt" (`DWRITE_FONT_AXIS_TAG_ITALIC` and `DWRITE_FONT_AXIS_TAG_SLANT`) axis values. 
+
+Another registered axis is "opsz" (`DWRITE_FONT_AXIS_TAG_OPTICAL_SIZE`). An optical font family such as
+Sitka includes fonts that differ along the "opsz" axis, meaning they are designed to be used at different
+point sizes. The WWS font family model does not have an optical size axis, so the Sitka font family must
+be split into multiple WWS font families: "Sitka Small", "Sitka Text", "Sitka Subheading", and so on. Each
+WWS font family corresponds to a different optical size, and it's left to the user to specify the right
+WWS family name for a given font size. With the typographic font family model, the user can simply choose
+"Sitka", and the application can automatically set the "opsz" axis value based on the font size.
+
+Advantages of the typographic font family model are:
+
+  - Fonts can be grouped into families as intended by the designer, rather than being split into
+    subfamilies due to limitations of the font family model.
+
+  - An application can automatically select the correct "opsz" axis value based on the font size
+    rather than exposing different optical sizes to the user as different font families.
+
+  - Arbitrary instances of variable fonts can be selected. For example, if a variable font supports
+    weights in the continuous range 100-900, the typographic model can select _any_ weight in this range.
+    The older font family models can only choose the nearest weight from among the named instances defined
+    by the font.
+
+However, migrating to the typographic font family model raises potential compatibility issues. The
+next section describes these issues and how they are mitigated by the hybrid font selection model.
+
+## Hybrid Font Selection Model
```

Comment:
The CSS spec https://www.w3.org/TR/css-fonts-4/#font-matching-algorithm documents in detail the algorithm at how one arrives at a match, in a model that is similar to WWS. Perhaps it would be helpful to specify as well how this new DWrite API finds the nearest match? 

Does the output have some kind of matching score? Would it be possible to specify priorities, i.e. it's more important to find an exact match for dimension m than for dimension n, or some such?

I am thinking about how this implementation could be used from the browser so that the output matches the behavior of what the CSS spec expects - or whether the CSS spec can only be implemented by building own data structures.

Reply:
For matching purposes, a "distance" is computed along each axis between the requested and actual value. Well-known axes have a multiplier that is applied distance in order to give behavior as close as possible to under the weight-stretch-style model.

That said, it's possible none of the newer APIs are appropriate for HTML engines to use because the CSS spec explicitly defines a font matching algorithm based on weight, stretch, and style. How the font family name is determined is not spelled out by the CSS spec, but it's kind of implied that it should be the weight-stretch-style name.